gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -302,15 +306,48 @@ def create_alt_bases_field_schema(self):
           type=bigquery_util.TableFieldConstants.TYPE_STRING,
           mode=bigquery_util.TableFieldConstants.MODE_NULLABLE,
           description='The ALT part of the annotation field.'))
-      for annotation_name in annotation_names:
+      gen_type_keys = self._type_keys_from_names(annotation_names, annot_field)
+      for annotation_name, type_key in zip(annotation_names, gen_type_keys):
+        if type_key in self._header_fields.infos:
+          vcf_type = self._header_fields.infos[type_key][
+              vcf_header_io.VcfParserHeaderKeyConstants.TYPE]
+        else:
+          vcf_type = vcf_header_io.VcfHeaderFieldTypeConstants.STRING
+          if self._infer_annotation_types:
+            logging.warning(('Annotation field %s has no corresponding header '
+                             'field with id %s to specify type. Using type %s '
+                             'instead.'), annotation_name, type_key, vcf_type)
         annotation_record.fields.append(bigquery.TableFieldSchema(
             name=bigquery_util.get_bigquery_sanitized_field(annotation_name),
-            type=bigquery_util.TableFieldConstants.TYPE_STRING,
+            type=bigquery_util.get_bigquery_type_from_vcf_type(vcf_type),
             mode=bigquery_util.TableFieldConstants.MODE_NULLABLE,
             description=annotation_descs.get(annotation_name, '')))
       alternate_bases_record.fields.append(annotation_record)
     return alternate_bases_record
 
+  def _type_keys_from_names(self, annotation_names, annot_field):
+    #  type: (List[str], str) -> str
+    for annotation_name in annotation_names:
+      type_key = vcf_header_io.BASE_TYPE_KEY.format(annot_field,
+                                                    annotation_name)
+      yield type_key
+
+  def gen_all_info_type_keys(self):
```

Comment:
Curious: why are you using generators instead of just returning a list/set? I personally find simple list/set return types to be easier to work with especially if you end up creating a list/set from these anyway.

Reply:
Great point! Thanks for the clarification!